Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,461 @@
+    MeasureTheory.tendsto_measure_iUnion Set.monotone_accumulate
+  rw [ENNReal.tendsto_atTop (measure_ne_top μ (⋃ n, Accumulate K n)), Set.iUnion_accumulate] at this
+  obtain ⟨N, hN⟩ := this ε hε
+  use N
+  simp_all only [Function.comp_apply, mem_Icc, le_refl]
+
+/-- For a finite measure `μ`, we can extract from a countable cover that has full measure, a finite
+cover of accumulated sets for which the complement has measure below `ε`. -/
+lemma exists_sub_le_measure_accumulate_compl [IsFiniteMeasure μ] (K : ℕ → Set α)
+    (hKmeas : ∀ n, MeasurableSet (K n)) (h : μ (⋃ n, K n) = μ Set.univ) {ε : ENNReal} (hε : ε > 0) :
+    ∃ n, μ ((Set.Accumulate K n)ᶜ) ≤ ε := by
+  obtain ⟨n, hn⟩ := exists_sub_le_measure_accumulate K h hε
+  use n
+  have hK2 : MeasurableSet (Set.Accumulate K n) := by
+    rw [Set.Accumulate]
+    measurability
+  rw [measure_compl hK2 (measure_ne_top μ _)]
+  exact tsub_le_iff_tsub_le.mp hn
+
+/-- For a finite measure `μ`, we can construct a dense sequence such that for any radius, we can
+cover all but a set of measure below `ε`. -/
+lemma exists_denseRange_sub_le_measure_univ_of_separableSpace [PseudoMetricSpace α]
+    [TopologicalSpace.SeparableSpace α] [Nonempty α] (μ : Measure α) [IsFiniteMeasure μ] :
+    ∃ u : ℕ → α, DenseRange u ∧
+    ∀ ε > 0, ∀ δ > 0, ∃ N : ℕ, μ (Set.univ) - ε ≤ μ (⋃ i ≤ N, Metric.ball (u i) δ) := by
+  obtain ⟨u, hu⟩ := TopologicalSpace.exists_dense_seq α
+  use u, hu
+  intro ε hε δ hδ
+  apply exists_sub_le_measure_accumulate (fun y ↦ Metric.ball (u y) δ) (?_) hε
+  apply le_antisymm (measure_mono fun ⦃a⦄ _ ↦ trivial)
+  exact measure_mono <| fun y _ => Set.mem_iUnion.mpr (DenseRange.exists_dist_lt hu y hδ)
+
+/-- For a finite measure `μ`, we can construct a dense sequence such that for any radius, we can
+find a `N` such that the measure of the complement of the first `N` balls of radius `1/(j+1)` is
+at most `ε/2^(j+1)`. -/
+lemma exists_denseRange_sub_le_measure_univ_of_separableSpace_compl [PseudoMetricSpace α]
+    [TopologicalSpace.SeparableSpace α] [Nonempty α] [OpensMeasurableSpace α] (μ : Measure α)
+    [IsFiniteMeasure μ] {ε : ENNReal} (hε : 0 < ε) : ∃ u : ℕ → α, DenseRange u ∧
+    ∀ j : ℕ, ∃ N : ℕ, μ ((⋃ i ≤ N, Metric.ball (u i) (1/(j+1)))ᶜ) ≤ ε/2^(j+1) := by
+  obtain ⟨u, hu, cover⟩ := exists_denseRange_sub_le_measure_univ_of_separableSpace μ
+  have hεj_pos : ∀ j : ℕ, ε/(2^(j+1)) > 0 :=
+    fun j => ENNReal.div_pos hε.ne' (Ne.symm (ne_of_beq_false rfl))
+  use u, hu
+  intro j
+  obtain ⟨N, hN⟩ := cover (ε/(2^(j+1))) (hεj_pos j) (1/(j+1)) (Nat.one_div_pos_of_nat)
+  use N
+  rw [measure_compl (by measurability) (measure_ne_top μ _)]
+  exact tsub_le_iff_tsub_le.mp hN
+
+/-- A measure `μ` is separable if there is a separable set `S` such that `μ S = μ Set.univ`. -/
+ def IsSeparable [TopologicalSpace α] (μ : Measure α) : Prop :=
+  ∃ S : Set α, TopologicalSpace.IsSeparable S ∧ μ S = μ Set.univ
```

Comment:
Tell me if I am wrong, but I feel like the condition should rather be
```math
\mu(S^c) = 0.
```
This is equivalent for finite measures but it seems more natural to me.

Reply:
This is the case at many places in the file, not only this definition.

Reply:
You are right, I've changed this (and it cleans up some proofs as well!)